relational-structs/src/relational_structs/spaces.py

```diff
@@ -82,6 +82,19 @@ def __init__(
         high = np.full(shape, np.inf, dtype=dtype)
         super().__init__(low, high, shape, dtype=dtype)
 
+    def update_constant_objects(self, constant_objects: list[Object]) -> None:
+        """Update the constant objects in this space.
+
+        This is particularly important if the number of objects or their
```

Comment:
hmmm this actually points to a bigger issue -- for typical RL / IL / ML approaches, we need constant vector dimensionality. I'm not sure yet what the easiest way to resolve this would be. let's discuss this in today's meeting